File: js/minCostClimbingStairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    minCosts = [0, 0]
    while(minCosts.length < cost.length + 1) {
        minCosts.push(Math.min(minCosts[minCosts.length - 1] + cost[minCosts.length - 1], 
                               minCosts[minCosts.length - 2] + cost[minCosts.length - 2]))
    }    
    return minCosts[minCosts.length-1]

};

File: min_cost_climbing_stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        cumulative = [0] * (len(cost) + 1)
        for i in range(2, len(cost)+ 1):
            cumulative[i] = min(cost[i-1] + cumulative[i-1], cost[i-2] + cumulative[i-2])
        #print(cumulative)
        return cumulative[-1] 
    
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        min_costs = [0] * (len(cost) + 1)
        #min_costs[1] = cost[0]
        for i in range(2, len(cost) + 1):
            min_costs[i] = min(min_costs[i-1] + cost[i-1], min_costs[i-2] + cost[i-2])
        return min_costs[-1]
        